### ISSUE TITLE
Function calls not hanging when over column width

### ISSUE THREAD

**Issue by JohnnyMorganz** [JohnnyMorganz/StyLua#368]
For the input below, I expected that the formatted output kept the same amount of lines:

```lua
foo.bar('-------------------------------------------------------------------------------------------------------------')
	.returns()
foo.bar('--------------------------------------------------------------------------------------------------')
	.returns()
	.way()
	.longer()
	.chain()
```

However, StyLua formatted it like the output below, where each function call chain is joined on a line. The lines are longer than the column limit of 120, and that doesn't read nice.

```lua
foo.bar("-------------------------------------------------------------------------------------------------------------").returns()
foo.bar("--------------------------------------------------------------------------------------------------").returns().way().longer().chain()
```

I'm posting it here, because it looks like a duplicate of this issue.

__Originally posted by @TjeuKayim in https://github.com/JohnnyMorganz/StyLua/issues/78#issuecomment-1033593230__

**Comment by matthargett** [comment]
Another angle, I'd expect this to act the same as the examples above:
```lua
foo.bar(--[[-------------------------------------------------------------------------------------------------------------]])
	.returns()
foo.bar(--[[--------------------------------------------------------------------------------------------------]])
	.returns()
	.way()
	.longer()
	.chain()
```

**Comment by TjeuKayim** [comment]
Another issue like this:

```lua
foo
    .bar('------------------------------------------------------')
    .returns('-----------------------------------------') -- some comment
```

This is wrapped on one line and then it exceeds the column limit because of the comment.

**Comment by JohnnyMorganz** [comment]
Out of interest, what kind of context do you have which chains function calls like this, rather than chained method calls (`:foo()`)?

When I originally implemented call hanging, I assumed that dot index `.foo()` calls would not be hanged, but I guess that's not true. Maybe I should just always hang function calls like we do method calls in all situations?

**Comment by matthargett** [comment]
I agree on special-casing `new` not being great. I think I like the consistency of the hanging more as well, even in this builder pattern case.

**Comment by JohnnyMorganz** [comment]
Incredibly sorry about the delay in sorting this out. Should be fixed in #470 and available in the next release
